src/filters/byte-format.js

```diff
@@ -0,0 +1,14 @@
+import { module } from 'angular';
+
+export default module('app.filters.byte-format', [])
+    .filter('byteFormat', () => (bytes, precision) => {
+      if (bytes === undefined) return '-';
+      if (bytes === 0) return '0 MB';
+      if (isNaN(parseFloat(bytes)) || !isFinite(bytes)) return '-';
```

Comment:
Seriously, the rest wouldn't work if it's not a number. Anyway, the `parseFloat` doesn't hurt since it's idempotent.